vault/rekey.go

```diff
@@ -350,6 +350,11 @@ func (c *Core) BarrierRekeyUpdate(key []byte, nonce string) (*RekeyResult, error
 	}
 
 	if len(c.barrierRekeyConfig.PGPKeys) > 0 {
+		if len(results.SecretShares) != len(c.barrierRekeyConfig.PGPKeys) {
```

Comment:
IIRC this isn't checked at command start time either. While it's good to check here, really the check here (and really, directly after line 339 above) needs to be whether the number of secret shares is equivalent to the rekey config's number...e.g. whether the shamir split was successful. If so, then if at init time we checked whether the rekey config's number of shares matches the number of PGP encrypted shares we should be good.

Note that the number of PGP keys actually needs to match the number of shares minus the number of stored shares.

Also, this same logic is in the recovery case so checks need to be put there too.